### DISCUSSION TITLE
Summary of ping and iperf3 results in my environment

### DISCUSSION THREAD

**Discussion by s5uishida** [edgecomllc/eupf#553] [General]
Hi,

Below is a quick summary of the results of trying ping and iperf3 using eUPF.

https://github.com/s5uishida/simple_confirmed_info_for_mobile_network#ping-and-iperf3

FYI.

**Comment by s5uishida** [comment]
There were no more **NG** results and I had reached the first milestone.

**Reply by pirog-spb** [reply]
Hi @s5uishida 

Good job!

When do you plan to test eUPF performance using native xdp mode? :)

-- BR Alex

**Reply by s5uishida** [reply]
Hi @pirog-spb

Sorry.

I couldn't get the equipment to verify it so I didn't do anything.
I would have liked to be able to verify it in my existing VirtualBox environment, but it didn't seem that easy.
I'd like to try this if I can get the equipment.

Best regards,

--Shigeru

**Reply by linouxis9** [reply]
> Hi @pirog-spb
> 
> Sorry.
> 
> I couldn't get the equipment to verify it so I didn't do anything. I would have liked to be able to verify it in my existing VirtualBox environment, but it didn't seem that easy. I'd like to try this if I can get the equipment.
> 
> Best regards,
> 
> --Shigeru

I'm still hopping to finalize the integration of [eUPF's eBPF code inside PacketRusher](https://github.com/HewlettPackard/PacketRusher/pull/47); it would help for the testing you do :-)
Sadly, it's a bit difficult to support native XDP Mode as there is no XDP hooks on TX yet, and I had not much more time to investigate further. In the current not native state, gtp5g is faster, but using gtp5g is not the best as it makes PacketRusher needs a N3/N2 IP/gNodeB per UE that needs a tunnel for max performance...

**Reply by s5uishida** [reply]
Hi @pirog-spb 

I tried native xdp mode on VMs on Proxmox VE, but the performance was not as good as I expected.

https://github.com/s5uishida/simple_measurement_of_upf_performance_3

I would be very grateful if you could give me some advice on tuning.

Best regards,

--Shigeru

**Reply by s5uishida** [reply]
Hi @pirog-spb 

Sorry.

I remembered the information you gave me before. I'll try it.

https://github.com/edgecomllc/eupf/discussions/507
